rust/lance/src/dataset.rs

```diff
@@ -136,6 +138,15 @@ pub struct ReadParams {
     pub session: Option<Arc<Session>>,
 
     pub store_options: Option<ObjectStoreParams>,
+
+    /// If present, dataset will use this to resolve the latest version
```

Comment:
But this is `ReadParams` so that raises a different question which is why a read operation would commit a new manifest?  I've updated the PR with a lengthier comment that hopefully addresses your concern.

Reply:
Ah I see. Nevermind.